From 93b9f41ae11a9adf84d2a1805b6a9c7b0d557832 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 11 Aug 2015 15:52:10 +0800
Subject: [PATCH 0169/5242] MLK-11340-43 usb: chipidea: imx: fix usb charger
 detection init NULL pointer

commit  43f0e787290f1c1512aecbfbb86ef4ba23c1a1c2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Dereference the charger->psy after the power_supply is created to fix the
null pointer issue.

Signed-off-by: Li Jun <b47624@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index 7efda33..34ecaae 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -409,7 +409,6 @@ static int imx_usb_charger_get_property(struct power_supply *psy,
 static int imx_usb_register_charger(struct usb_charger *charger,
 		const char *name)
 {
-	struct power_supply		*psy = charger->psy;
 	struct power_supply_desc	*desc = &charger->psy_desc;
 
 	if (!charger->dev)
@@ -427,15 +426,16 @@ static int imx_usb_register_charger(struct usb_charger *charger,
 	desc->properties	= imx_usb_charger_power_props;
 	desc->num_properties	= ARRAY_SIZE(imx_usb_charger_power_props);
 	desc->get_property	= imx_usb_charger_get_property;
-	psy->supplied_to	= imx_usb_charger_supplied_to;
-	psy->num_supplicants	= sizeof(imx_usb_charger_supplied_to)
-					/ sizeof(char *);
 
-	charger->psy = devm_power_supply_register(charger->dev->parent,
+	charger->psy = devm_power_supply_register(charger->dev,
 						&charger->psy_desc, NULL);
 	if (IS_ERR(charger->psy))
 		return PTR_ERR(charger->psy);
 
+	charger->psy->supplied_to	= imx_usb_charger_supplied_to;
+	charger->psy->num_supplicants	= sizeof(imx_usb_charger_supplied_to)
+					/ sizeof(char *);
+
 	return 0;
 }
 
-- 
1.7.9.5

